### PR TITLE
AIP-46: New modules for ElGamal, Pedersen and Bulletproofs over Ristretto255

### PR DESCRIPTION
**Pull request by mstraka100**

### CODE REVIEW

File: aips/ristretto_bulletproofs_elgamal.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+The ElGamal encryption and Pedersen commitment modules are implemented without Move natives over the Ristretto255 Move module referenced above. 
+
+## Risks and Drawbacks
+
+Future obsolescence is often a risk when implementing cryptography primitives. As of writing this, both Ristretto255 and Bulletproofs have been in use for several years, making their continued used in the immediate future more likely than not. ElGamal encryption and Pedersen commitments have been in use for multiple decades.
+## Timeline
+
+### Suggested developer platform support timeline
+
+Describe the plan to have SDK, API, CLI, Indexer support for this feature is applicable.
```
Comment (davidiw) [inline aips/ristretto_bulletproofs_elgamal.md:59]: Is this applicable? I suspect not really.

**Review by davidiw — COMMENTED** [review]
Looking good. Could you please provide links to the wikipedia entries or something to educate the user of what these are and there relevant implementations, such as the paper that introduced them or some recent book. Please use AIP number 46. @thepomeranian

**Comment by davidiw** [comment]
Also are we leveraging any nuanced aspects of them? For example, we have some changes to ed25519 for enhanced security.

**Comment by mstraka100** [comment]
@davidiw Updated the AIP with your feedback. I wouldn't say we're using any of the protocols in a nuanced way beyond their original design.

File: aips/ristretto_bulletproofs_elgamal.md (hunk 1 above)
Comment (alinush) [reply aips/ristretto_bulletproofs_elgamal.md]: It is not. Will clarify.

**Review by alinush — APPROVED** [review]
Thank you for putting this together! 🙏  I made several changes to improve the organization, add more details, and add context regarding the questions that each section seeks to answer.

**Comment by mstraka100** [comment]
Made some small edits and moved from draft to open. @davidiw should be good to get merged unless there's more feedback

**Comment by alinush** [comment]
Can we merge this @thepomeranian?

**Review by thepomeranian — APPROVED** [review]

**Pull request merged**